pkg/scheduler/plugins/util/nodelock/nodelock.go

```diff
@@ -84,18 +76,37 @@ func setNodeLock(nodeName string, lockName string) error {
 			klog.ErrorS(err, "Failed to get node when retry to update", "node", nodeName)
 			continue
 		}
-		newNode := node.DeepCopy()
-		newNode.ObjectMeta.Annotations[lockName] = time.Now().Format(time.RFC3339)
+		newNode = node.DeepCopy()
+		updateFunc(newNode.ObjectMeta.Annotations)
 		_, err = kubeClient.CoreV1().Nodes().Update(ctx, newNode, metav1.UpdateOptions{})
 	}
+	if err != nil {
+		return fmt.Errorf("failed to update node %s, exceeded retry count %d", nodeName, MaxLockRetry)
+	}
+	return nil
+}
+
+func setNodeLock(nodeName string, lockName string) error {
+	ctx := context.Background()
+	node, err := kubeClient.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if _, ok := node.ObjectMeta.Annotations[lockName]; !ok {
```

Comment:
This has changed the origin code meaning.

Reply:
ya, i should use this , It will not change the original logic
```go 
func setNodeLock(nodeName string, lockName string) error {
	ctx := context.Background()
	node, err := kubeClient.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
	if err != nil {
		return err
	}


         // change.....
	if _, ok := node.ObjectMeta.Annotations[lockName]; ok {
		klog.V(3).Infof("node %s is locked", nodeName)
		return fmt.Errorf("node %s is locked", nodeName)
	}
	updateFunc := func(annotations map[string]string) {
		annotations[lockName] = time.Now().Format(time.RFC3339)
	}
	err = updateNodeAnnotations(ctx, node, updateFunc)
	if err != nil {
		return fmt.Errorf("setNodeLock exceeds retry count %d", MaxLockRetry)
	}
	klog.InfoS("Node lock set", "node", nodeName)
	return nil
}

```

Reply:
ya, i should use this , It will not change the original logic
```go
func setNodeLock(nodeName string, lockName string) error {
	ctx := context.Background()
	node, err := kubeClient.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
	if err != nil {
		return err
	}


         // change.....
	if _, ok := node.ObjectMeta.Annotations[lockName]; ok {
		klog.V(3).Infof("node %s is locked", nodeName)
		return fmt.Errorf("node %s is locked", nodeName)
	}
	updateFunc := func(annotations map[string]string) {
		annotations[lockName] = time.Now().Format(time.RFC3339)
	}
	err = updateNodeAnnotations(ctx, node, updateFunc)
	if err != nil {
		return fmt.Errorf("setNodeLock exceeds retry count %d", MaxLockRetry)
	}
	klog.InfoS("Node lock set", "node", nodeName)
	return nil
}

```

Reply:
Current change is ok. The former returns nil when lock not exist. That's wrong.

Reply:
thanks for catching that！